crypto/objects/objects.txt

```diff
@@ -385,6 +385,11 @@ sm-scheme 301           : SM2                   : sm2
 sm-scheme 401           : SM3                   : sm3
 sm-scheme 504           : RSA-SM3		: sm3WithRSAEncryption
 
+# ZUC stream cipher
+sm-scheme 201           : ZUC                   : zuc
```

Comment:
@InfoHunter Is there such a thing as ZUC as an actual stream cipher, and not just a keystream generator? (Standards wise.)

I googled around, but not much in English.

I'm asking because in that case, you could make an argument for it's own EVP interface. Exposing only EEA3 to EVP seems curious to me, since the use case is so restricted and so different from a "normal" stream cipher. (Even more weird if you have planned efforts for EIA3, since the maximum message length is slightly below 64KB due to mobile network details.)

Reply:
Hi there, this is a good question and what you concerned is exactly the thing what I care about...

Unfortunately, there is no standard defined *yet* describing a stream cipher other than EEA3. Even in Chinese national standards, ZUC is also only used as a keystream generator. This is the *current* status.

Fortunately, we (some Chinese companies) are putting lots of effort to make an actual ZUC stream cipher, which could be used in a TLS context (for instance the IV is not formed by those very specific parameters). But we are not the agency in control of this kind of things, so we need to coordinate with the 'real' agency (the Chinese SCA) to push this forward.

I hope this could happen in a not very far future...